library/src/main/java/io/ona/kujaku/interfaces/IKujakuMapView.java

```diff
@@ -257,4 +261,59 @@
     boolean isKujakuLayerAdded(@NonNull KujakuLayer kujakuLayer);
 
     void removeLayer(@NonNull KujakuLayer kujakuLayer);
+
+
+    /**
+     * Start the instance of TrackingService that record filtered locations regarding the given {@link TrackingServiceOptions}
+     * You need to register a {@link TrackingServiceListener} to receive notifications as
+     * - {@link TrackingServiceListener#onFirstLocationReceived(Location)}
+     * - {@link TrackingServiceListener#onNewLocationReceived(Location)}
+     * - etc.
+     * The Class arguments is used to instantiate an activity when clicking the service notification
+     *
+     * @param context
+     * @param cls
+     * @param trackingServiceListener
+     * @param options
+     */
+    void startTrackingService(@NonNull Context context, @NonNull Class<?> cls, @NonNull TrackingServiceListener trackingServiceListener, TrackingServiceOptions options);
+
+    /**
+     * Stop the instance of TrackingService
+     * It returns a collection of recorded Locations
+     *
+     * @param context
+     * @return
+     */
+    List<Location> stopTrackingService(@NonNull Context context);
+
+    /**
+     * Unbind from the TrackingService instance to be able to stop the service
```

Comment:
1. At this point our service life is not controlled by the `binding` idea
2. Add the `TrackingService` link to the **Table of Contents**
3. I would like to know how much you have tested the code in the field using the **Kujaku sample app** probably moving for more than 1 km. I am having unexpected results

Reply:
What kind of unexpected results ? i have tested the kujaku sample multiple times during the itinerary between office and home and the passive activity shows me the exact route i have driven.

Reply:
I walked out of the office to a Cafe and left the app as the foreground application and put it in my pocket. It recorded half my route. The route is recorded halfway there and from that point back.

Device is Android version 8.1.0

What android version did you test with?

Reply:
Ok,There are some Android settings to modify :
- You need to whitelist the sample application from the battery optimization. This is mandatory.
- On some Huawei phones on Android 8.1, there is a new setting to modify : Settings->Battery->Launch->Managed Manually: Set the 3 Auto-launch, Secondary Launch, Run in Background

I tested the app with Android 8.1, Android 5.1 . I will test it on Android 7.0 and 6.0

Reply:
Thanks, this worked and it recorded a 2 and a half hour. Can you create that as an issue and we can also document it later for easy reference and troubleshooting? The issue will basically state that this has to be done manually and that documentation for the same has to be done while we work on adding the ability to prompt the user to enable this

Reply:
ok make sense